dvc/output.py

```diff
@@ -843,7 +849,8 @@ def get_dir_cache(self, **kwargs):
         try:
             objects.check(self.odb, obj)
         except FileNotFoundError:
-            self.repo.cloud.pull([obj.hash_info], **kwargs)
+            remote = self.repo.cloud.get_remote_odb(self.remote)
+            self.repo.cloud.pull([obj.hash_info], odb=remote, **kwargs)
```

Comment:
We need support for better error messages, like where the remote has been specified in case of `NoRemoteError` or `RemoteNotFound`.